presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -378,7 +378,7 @@ public static Boolean stIsClosed(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     @SqlType(BOOLEAN)
     public static Boolean stIsEmpty(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        return deserialize(input).isEmpty();
+        return deserializeEnvelope(input) == null;
```

Comment:
It took me some time to follow the code. 

It seems like it was a bad design decision to return `null` from `deserializeEnvelope`. 

Currently it is not clear what does the `null` mean. The method returns `null` if the input Slice is empty, or if the envelope is empty. However, for the `ENVELOPE` type of geometry it returns an empty envelope, and not null.

My proposal is to return `null` (or better `Optional.empty()`) if the input is empty. And return and empty envelope instead of null if the envelope is empty. But given the number of method where the `deserializeEnvelope` is used it should be done as a separate PR.